MapboxDirections/MBVisualInstruction.swift

```diff
@@ -41,6 +41,18 @@ open class VisualInstruction: NSObject, NSSecureCoding {
      */
     @objc public let secondaryTextComponents: [VisualInstructionComponent]?
     
+    /**
+     :nodoc:
+     The maneuver type for the `VisualInstruction`.
+     */
+    @objc public var maneuverType: ManeuverType
+    
+    /**
+     :nodoc:
+     The modifier type for the `VisualInstruction`.
+     */
+    @objc public var maneuverDirection: ManeuverDirection
```

Comment:
Soon the secondary and primary components will each have type and modifier. How should these properties be renamed?

`primaryManeuverType` and `secondaryManeuverType`?

Reply:
How about wrapping primary and secondary in an instructions object of its own?
Not sure about the name but something along these lines:
```swift
instructions.primary.maneuverType
instructions.secondary.maneuverType
```

Reply:
Does this make sense even though this property is an array of VisualInstructionsComponent?

https://github.com/mapbox/MapboxDirections.swift/blob/fc3e74bdf00e0eb1bf550e731fd5a1b00cfe9c9a/MapboxDirections/MBVisualInstruction.swift#L22-L28

Would each component have a type + modifier then?

Reply:
I think the data structure is already uncomfortably deeply nested. It might be a different story if we were working purely with Swift structs or tuples, but classes have a certain amount of mental overhead. How about `primaryManeuverType` and `secondaryManeuverType`, same for modifiers?